pyswarms/utils/functions/grid_search.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+
+"""
+Hyperparameter grid search.
+
+Compares the relative performance of hyperparameter value combinations in
+reducing a specified objective function.
+
+For each hyperparameter, user can provide either a singe value or a list
+of possible values, and their cartesian product is taken to produce a grid
+of all possible combinations. These combinations are then tested to produce
+a list of objective function scores. The default of the optimize method
+returns the hyperparameters that yield the minimum score, yet maximum score
+can also be assessed.
+"""
+
+import operator as op
+import itertools
+import numpy as np
+
+class GridSearch():
```

Comment:
I think it would be better to use the new-style introduced in Python 2.2 for declaring classes.

```python
# Old-style
class GridSearch:
    pass

# New-style
class GridSearch(object):
    pass
```

Reply:
Yes, I agree.